lib/bots/notifybot.ts

```diff
@@ -762,35 +767,44 @@ export class NotifyBot extends ProcBot {
 			// therefore is now implemented. It is implicitly an incrementing version.
 			if (!component.oldVersion) {
 				component.incrementing = true;
-			} else if (component.oldVersion > component.newVersion) {
+			} else if (semverGt(component.oldVersion, component.newVersion)) {
 				// A new version less than an older version is a version down.
 				component.incrementing = false;
-			} else if (component.newVersion > component.oldVersion) {
+			} else if (semverGt(component.newVersion, component.oldVersion)) {
 				component.incrementing = true;
 			}
 
 			// If there is an old component, use it to cull anything before that version.
+			// We end up with a portion of the changelog that only includes details
+			// between the old version and the last entry in the changelog.
 			if (component.oldVersion) {
-				changeParts = changeParts[0].split(component.incrementing ? component.oldVersion :
-					component.newVersion);
+				changeParts = changeParts[0].split(component.incrementing ? `## ${component.oldVersion}` :
+					`## ${component.newVersion}`);
 			}
-			// We are guaranteed a new version, as it's already been verified.
+
+			// The splitting strips the previous header entry, so add it back depending on whether we're
+			// incrementing or decrementing in version.
+			// Then split the changelog portion again to end up only with a portion that
+			// contains all entries between the old and new version.
 			changelogString = `## ${component.incrementing ? component.newVersion : component.oldVersion}` +
 				changeParts[0].split(component.incrementing ? component.newVersion : (component.oldVersion || ''))[1];
 
-			// We need to search the first entry in the array and split on every version
-			// so we know which versions to report back.
+			// Split the changelog based on version entries. This ensures we end up with
+			// an array, each entry being a single version set of changes, for example:
+			//  "v.1.2.3 - 2018-08-02\n\n* This does something else #124 [Author]"
+			//  "v.1.2.2 - 2018-08-01\n\n* This does something #123 [Author]"
```

Comment:
There's an extra `.` in these version numbers, between the `v` and the `1`

Reply:
Fixed.